loki/backend/fgencon.py

```diff
@@ -0,0 +1,207 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+from loki.backend.fgen import FortranCodegen
+from loki.frontend.source import SourceStatus
+
+
+__all__ = ['FortranCodegenConservative']
+
+
+class FortranCodegenConservative(FortranCodegen):
+    """
+    Strictly conservative version of :any:`FortranCodegen` visitor
+    that will attempt to use existing :any:`Source` information from
+    the frontends where possible.
+    """
+
+    def visit_Node(self, o, *args, **kwargs):
+        if o.source and o.source.status == SourceStatus.VALID:
+            return o.source.string
```

Comment:
```py
if o.source and o.source.status == SourceStatus.VALID:
    return o.source.string
```

is a common pattern in `fgencode`.
Wouldn't it make sense to wrap that in e.g.,

```py
def return_valid_source(self, o):
     if o.source and o.source.status == SourceStatus.VALID:
        return o.source.string
```

Reply:
Unfortunately no, since the early return is what prevents the rest of the method to be executed. If you wrap this in a utility you lose the early return and have to add another conditional around the utility, which defeats the point for a two-line snippet.